packages/fxa-settings/src/lib/hooks/useRedirect/index.tsx

```diff
@@ -0,0 +1,54 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { isAllowed } from 'fxa-shared/configuration/convict-format-allow-list';
+import {
+  BaseIntegrationData,
+  IntegrationType,
+  isWebIntegration,
+  useConfig,
+  useFtlMsgResolver,
+} from '../../../models';
+import { useLocation } from '@reach/router';
+import {
+  AuthUiErrors,
+  composeAuthUiErrorTranslationId,
+} from '../../auth-errors/auth-errors';
+
+export interface UseRedirectIntegration {
+  type: IntegrationType;
+  data: {
+    redirectTo: BaseIntegrationData['redirectTo'];
+  };
+}
+
+/*
+ * Check if the integration contains a valid `redirectTo` based on
+ * a whitelist maintained in the config.
+ *
+ * At the time of writing, this is only valid for web integrations.
+ * OAuth integrations must derive keys before redirecting.
+ */
+export function useRedirect(integration: UseRedirectIntegration) {
```

Comment:
Why is this a good candiate for a hook? Not saying it's not. Just curious what your thinking hear is.

Reply:
At first I didn't have it as one, but we are going to need this logic for at least Signin as well. I pulled this out because it cleaned up that big function a bit and we can use it similarly over there.